Extension/src/Debugger/debugAdapterDescriptorFactory.ts

```diff
@@ -9,32 +9,51 @@ import * as path from 'path';
 import * as os from 'os';
 
 // Registers DebugAdapterDescriptorFactory for `cppdbg` and `cppvsdbg`. If it is not ready, it will prompt a wait for the download dialog.
-// Note: util.extensionContext.extensionPath is needed for the commands because VsCode does not support relative paths for adapterExecutableComand
+// NOTE: This file is not automatically tested.
 
-export class CppdbgDebugAdapterDescriptorFactory implements vscode.DebugAdapterDescriptorFactory {
+abstract class DebugAdapterDescriptorFactoryWithContext implements vscode.DebugAdapterDescriptorFactory {
+    protected readonly context: vscode.ExtensionContext;
+
+    // This is important for the Mock Debugger since it can not use src/common
+    constructor(context: vscode.ExtensionContext) {
+        this.context = context;
+    }
+
+    abstract createDebugAdapterDescriptor(session: vscode.DebugSession, executable: vscode.DebugAdapterExecutable | undefined): vscode.ProviderResult<vscode.DebugAdapterDescriptor>;
+}
+
+export class CppdbgDebugAdapterDescriptorFactory extends DebugAdapterDescriptorFactoryWithContext {
     public static DEBUG_TYPE : string = "cppdbg";
 
+    constructor(context: vscode.ExtensionContext) {
+        super(context);
+    }
+
     createDebugAdapterDescriptor(session: vscode.DebugSession, executable: vscode.DebugAdapterExecutable | undefined): vscode.ProviderResult<vscode.DebugAdapterDescriptor> {
         return util.isExtensionReady().then(ready => {
             if (ready) {
-                let command: string = path.join(util.extensionContext.extensionPath, './debugAdapters/OpenDebugAD7');
+                let command: string = path.join(this.context.extensionPath, './debugAdapters/OpenDebugAD7');
```

Comment:
Is the `context` new?

Reply:
Its the same context that is set in common.ts (util.extensionContext)